src/main/java/com/mentoring/ecommerce/adapter/out/persistence/postgresql/product/ProductPostgreSQLAdapter.java

```diff
@@ -0,0 +1,28 @@
+package com.mentoring.ecommerce.adapter.out.persistence.postgresql.product;
+
+import com.mentoring.ecommerce.application.port.out.FindProductPort;
+import com.mentoring.ecommerce.application.port.out.SaveProductPort;
+import com.mentoring.ecommerce.application.port.out.UpdateProductPort;
+import com.mentoring.ecommerce.domain.Product;
+import lombok.RequiredArgsConstructor;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+// @PersistenceAdaper
+public class ProductPostgreSQLAdapter implements FindProductPort, UpdateProductPort, SaveProductPort {
+
+    ProductPostgreSQLRepository repository;
+
+    @Override public List<Product> findAll() {
+        return (List<Product>) repository.findAll();
+    }
+
+    @Override public Product saveProduct(final Product product) {
+        return repository.save(product);
+    }
+
+    @Override public Product updateProduct(final Product product, final Integer id) {
```

Comment:
@Pelicer 
I'm not sure how would `updateProduct` be implemented here in the `repository`, as its function is pretty similar to `saveProduct`.

Reply:
It's actually the same thing. Calling `saveProduct` when the object you pass to it already exists on the database will cause the repository to update it, instead of creating another entry.